source/frontend/src/features/mapSideBar/management/tabs/activities/edit/ContactListForm.tsx

```diff
@@ -0,0 +1,59 @@
+import { ArrayHelpers, FieldArray, FormikProps, getIn } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+
+import { LinkButton } from '@/components/common/buttons';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import { RestrictContactType } from '@/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent';
+import { IContactSearchResult } from '@/interfaces';
+
+import { ManagementActivityFormModel } from './models';
+
+export interface IContactListForm {
+  field: string;
+  contactType: RestrictContactType;
+  formikProps: FormikProps<ManagementActivityFormModel>;
+}
+
+export const ContactListForm: React.FunctionComponent<IContactListForm> = ({
+  field,
+  contactType = RestrictContactType.ALL,
+  formikProps,
+}) => {
+  // clear out existing values instead of removing last item from array
+  const onRemove = (array: Array<any>, index: number, arrayHelpers: ArrayHelpers) => {
+    if (array.length > 1) {
+      arrayHelpers.remove(index);
+    } else {
+      arrayHelpers.replace(index, null);
+    }
+  };
+
+  const contacts = getIn(formikProps.values, field) as IContactSearchResult[];
+  return (
+    <FieldArray name={field}>
+      {arrayHelpers => (
+        <>
+          {contacts.map((_, index) => (
+            <Row key={`${field}.${index}`} className="no-gutters pr-4 mr-2">
```

Comment:
this is probably copied, but is still "new" code, therefore, we shouldn't be using indexes to build keys, prefer a unique identifier and fall back to index if unavailable.

Reply:
Updated